service/history/api/recordactivitytaskstarted/api.go

```diff
@@ -154,12 +183,22 @@ func recordActivityTaskStarted(
 		if ai.RequestId == requestID {
 			response.StartedTime = ai.StartedTime
 			response.Attempt = ai.Attempt
-			return response, false, nil
+			return response, activity_accepted, nil
 		}
 
 		// Looks like ActivityTask already started as a result of another call.
 		// It is OK to drop the task at this point.
-		return nil, false, serviceerrors.NewTaskAlreadyStarted("Activity")
+		return nil, rejected_with_error, serviceerrors.NewTaskAlreadyStarted("Activity")
+	}
+
+	activityModified, err := processActivityWorkflowRules(shardContext, mutableState, ai)
+	if err != nil {
+		return nil, rejected_with_error, err
+	}
+
+	if activityModified && ai.Stamp != request.Stamp {
```

Comment:
what if after modifying the activity the `Stamp` in `ai` matches that in the `request`? 

I don't a stamp check before or inside `processActivityWorkflowRules`